src/uk/gov/defra/ffc/ADO.groovy

```diff
@@ -0,0 +1,46 @@
+package uk.gov.defra.ffc
+
+class ADO implements Serializable {
+
+  static void triggerPipeline(def ctx, String namespace, String chartName, String chartVersion, Boolean hasDatabase) {
+    if (hasDatabase) {
+      triggerDatabasePipeline(ctx, chartName, chartVersion)
+    }
+    triggerHelmPipeline(ctx, namespace, chartName, chartVersion)
+  }
+
+  static void triggerDatabasePipeline(def ctx, String database, String version) {
+    ctx.echo "Triggering ADO Database pipeline for ${database} ${version}"
+    String pipelineId = ctx.ADO_DATABASE_PIPELINE_ID
+    def data = "'{\"templateParameters\": {\"database\":\"$database\",\"tagValue\":\"$version\"}}'"
+    triggerBuild(ctx, pipelineId, data, database, version)
+  }
+
+  static void triggerHelmPipeline(def ctx, String namespace, String chartName, String chartVersion) {
+    ctx.echo "Triggering ADO Helm pipeline for ${chartName} ${chartVersion} in ${namespace}"
+    String pipelineId = ctx.ADO_HELM_PIPELINE_ID
+    def data = "'{\"templateParameters\": {\"namespace\": \"$namespace\",\"chart\":\"$chartName\",\"chartVersion\":\"$chartVersion\"}}'"
+    triggerBuild(ctx, pipelineId, data, chartName, chartVersion)
+  }
+
+  static void triggerBuild(def ctx, pipelineId, data, repository, version) {
+    ctx.withCredentials([
+      ctx.usernamePassword(credentialsId: 'ado-credentials', usernameVariable: 'username', passwordVariable: 'password')
+    ]) {
+      String buildId = ctx.sh(returnStdout: true, script: "curl -u $ctx.username:$ctx.password https://dev.azure.com/defragovuk/DEFRA-FFC/_apis/pipelines/${pipelineId}/runs?api-version=6.0-preview.1 -H 'Content-Type: application/json' -d $data | jq '.id'").trim()
```

Comment:
At this point, if build fails, how do we know without manual intervention/monitoring?

Reply:
All failed builds in Jenkins are reported to `ffc-notifications` slack workspace channel.

If the subsequent ADO pipeline fails that's not a concern of Jenkins.  That's a missing feature in our current ADO setup and should be addressed there.